middlewares/search.middleware.js

```diff
@@ -0,0 +1,27 @@
+"use strict";
+const Services = {
+    Search: require("../services/search.service")
+};
+
+/**
+ * @function parseQuery
+ * @param {JSON} req
+ * @param {JSON} res
+ * @param {JSON} next
+ * @description parses the json in the parameter
+ */
+function parseQuery(req, res, next){
+    let query = req.query.q;
```

Comment:
You might want to grab the query from req.body-- middleware.parseBody will put the validated query 'q' into there. Let's chat about this